src/python/tests/fasta/test_chunk.py

```diff
@@ -0,0 +1,441 @@
+        expectation: A context manager with expected exception (`pytest.raises` or nullcontext)
+    """
+
+    with expectation:
+        FastaChunking.check_chunk_size_and_tolerance(chunk_size, chunk_tolerance)
+
+
+@pytest.mark.parametrize(
+    "seq, pattern, expectation",
+    [
+        ("", None, [len("")]),
+        ("", re.compile("N"), [0]),
+        ("CANNAN", None, [len("NANNAN")]),
+        ("CANNAN", None, [6]),
+        ("CAAAAN", re.compile("N"), [6]),
+        ("NAAAAN", re.compile("N"), [1, 6]),
+        ("NANNAN", re.compile("N"), [1, 3, 4, 6]),
+        ("NANNAN", re.compile("NN"), [4, 6]),
+        ("NAAAAN", re.compile("NN"), [6]),
+    ],
+)
+def test_split_seq_by_n(seq: str, pattern: Optional[re.Pattern], expectation: list[int]) -> None:
+    """Tests the `chunk.split_seq_by_n` function.
+
+    Args:
+        seq: A sequence to split
+        pattern: A pattern to split on
+        expectation: A list of open chunk ends (like for python list slices)
+    """
+    assert FastaChunking.split_seq_by_n(seq, pattern) == expectation
+
+
+@pytest.mark.parametrize(
+    "chunk_ends, chunk_size, tolerated_size, expectation",
+    [
+        (None, 1, None, None),
+        ([], 1, None, []),
+        ([], 1, 1, []),
+        ([6], -1, None, [6]),
+        ([6], 6, None, [6]),
+        ([6], 5, None, [5, 6]),
+        ([4, 6], 0, None, [4, 6]),
+        ([4, 6], 1, None, [1, 2, 3, 4, 5, 6]),
+        ([4, 6], 2, None, [2, 4, 6]),
+        ([4, 6], 2, 4, [4, 6]),
+        ([1, 3, 4, 6], 2, 4, [1, 3, 4, 6]),  # do not join back
+        ([1, 3, 4, 6], 6, 0, [1, 3, 4, 6]),  # do not join back
+    ],
+)
+def test_split_seq_by_chunk_size(
+    chunk_ends: list[int], chunk_size: int, tolerated_size: Optional[int], expectation: list[int]
+):
+    """Tests the `chunk.split_seq_by_chunk_size` function.
+
+    Args:
+        chunk_ends: A list of chunk_ends (open ones, like for python list slices)
+        chunk_size: Chunk size
+        tolerated_size: A more relaxed value of the chunk size
+        expectation: A list of open chunk ends (python slices coordinates)
+    """
+    assert FastaChunking.split_seq_by_chunk_size(chunk_ends, chunk_size, tolerated_size) == expectation
+
+
+def test_individual_file_opener(tmp_path: Path) -> None:
+    """Tests the `chunk._individual_file_opener` function.
+
+    Args:
+        tmp_path: Where temporary files will be created.
+    """
+    test_dir = Path(tmp_path, "file_opener_test")
+    test_dir.mkdir()
+    test_path = Path(test_dir, "test.file")
+
+    with FastaChunking._individual_file_opener(str(test_path)) as out:
+        print("test", file=out)
+        print("test", file=out)
+    assert test_path.read_text(encoding="utf-8") == "test\ntest\n"
+
+    with FastaChunking._individual_file_opener(str(test_path)) as out:
+        print("test", file=out)
+    assert test_path.read_text(encoding="utf-8") == "test\n"
+
+    assert len(list(test_dir.iterdir())) == 1
```

Comment:
Is this line meant to check if the file has only been created once? Not sure if it is required (this is handled by the file system, not your code).